doc/sphinx/notebooks/advanced/varying_noise.pct.py

```diff
@@ -24,284 +24,314 @@
 
 # %%
-likelihood = gpflow.likelihoods.SwitchedLikelihood(
-    [gpflow.likelihoods.Gaussian(variance=1.0), gpflow.likelihoods.Gaussian(variance=1.0)]
+model = gf.models.GPR(
+    data=(X, Y_mean),
+    kernel=gf.kernels.SquaredExponential(),
+)
+gf.optimizers.Scipy().minimize(
+    model.training_loss, model.trainable_variables, options=optimizer_config
 )
-kernel = gpflow.kernels.Matern52(lengthscales=0.5)
-model = gpflow.models.VGP((X, Y_data), kernel=kernel, likelihood=likelihood, num_latent_gps=1)
+print_summary(model)
 
-set_trainable(model.q_mu, False)
-set_trainable(model.q_sqrt, False)
+# %%
+mean, var = model.predict_f(X_plot)
+plot_distribution(X_flat, Y_flat, X_plot, mean.numpy(), var.numpy())
 
-for _ in range(reduce_in_tests(1000)):
-    natgrad.minimize(model.training_loss, [(model.q_mu, model.q_sqrt)])
-    adam.minimize(model.training_loss, model.trainable_variables)
 
 # %% [markdown]
-# ### Plotting the fitted model
+# ### Create custom function for the noise variance
 #
-# Now that the noise variances have been estimated, we can see the final model fit.
-# The predictive variance is higher on the left side of the plot, where we know that the data have different variance.
-# We'll plot the known underlying function in green to see how effectively we've recovered the ground truth.
-# We can also print the model to examine the estimated noise variances:
+# We're modelling the `Y_mean` and its standard error is `Y_var / n_repeats`. We will create a simple function that ignores its input and returns these values as a constant. This will obviously only work for the `X` that corresponds to the `Y` we computed the variance of - that is good enough for model training, but notice that it will not allow us to use `predict_y`.
 
 # %%
-# let's do some plotting!
-xx = np.linspace(-5, 5, 200)[:, None]
+class FixedVarianceOfMean(gf.functions.Function):
+    def __init__(self, Y):
+        n_repeats = Y.shape[-1]
+        self.var_mean = np.var(Y, axis=-1, keepdims=True) / n_repeats
 
-mu, var = model.predict_f(xx)
+    def __call__(self, X):
+        return self.var_mean
 
-fig, ax = plt.subplots(1, 1, figsize=(12, 6))
-ax.plot(xx, mu, "C0")
-ax.plot(xx, mu + 2 * np.sqrt(var), "C0", lw=0.5)
-ax.plot(xx, mu - 2 * np.sqrt(var), "C0", lw=0.5)
 
-ax.plot(X, Y, "C1x", mew=2)
-ax.set_xlim(-5, 5)
-_ = ax.plot(xx, 2.5 * np.sin(6 * xx) + np.cos(3 * xx), "C2--")
+# %% [markdown]
+# Now we can plug that into our model:
 
 # %%
-from gpflow.utilities import print_summary
+model = gf.models.GPR(
+    data=(X, Y_mean),
+    kernel=gf.kernels.SquaredExponential(active_dims=[0]),
+    likelihood=gf.likelihoods.Gaussian(variance=FixedVarianceOfMean(Y)),
+)
+gf.optimizers.Scipy().minimize(
+    model.training_loss, model.trainable_variables, options=optimizer_config
+)
+print_summary(model)
 
-print_summary(model, fmt="notebook")
+# %%
+mean, var = model.predict_f(X_plot)
+plot_distribution(X_flat, Y_flat, X_plot, mean.numpy(), var.numpy())
```

Comment:
Maybe you could plot some error bars that include the calculated empirical variance of Y, and clarify why the smooth predictive variance are no longer containing the data between these error-bars (because we are predicting f)?

Reply:
Done.